### PR TITLE
Add Firebase Storage integration for image management

### LINKED ISSUE #150: Implement backend for images
### Overview
Integrate app with Firebase Storage to be able to store photos and previously participated in sessions. 
This is mainly used for "heavy" data, the integration should be optimized in terms of performance.

### PR DESCRIPTION
**Pull request by AntoineBastide47**
## Summary
   - Implements `ImageRepository` with comprehensive image storage and retrieval functionality
   - Adds Firebase Storage integration for managing account profile pictures, shop photos, space renter photos, and discussion images
   - Includes image compression and WebP encoding for optimized storage
   - Implements local caching mechanism to reduce redundant network calls
   - Adds extensive test coverage with 489 lines of integration tests

   ## Changes
   ### Core Implementation
   - **ImageRepository.kt**: New repository handling all image operations with Firebase Storage
     - Account profile picture save/load
     - Shop photos (multiple images) save/load
     - Space renter photos save/load
     - Discussion profile pictures save/load
     - Discussion photo messages (with unique IDs) save/load
     - Image compression with configurable quality and max dimensions
     - Local file caching for improved performance
     - Parallel loading/saving for bulk operations

   ### Configuration & Dependencies
   - Updated `minSdk` from 29 to 30 (required for ImageDecoder API)
   - Added Firebase Storage dependency
   - Configured Storage emulator on port 9199
   - Updated Storage security rules

   ### Testing Infrastructure
   - **ImageRepositoryTests.kt**: Comprehensive test suite covering:
     - All image save/load operations
     - Caching behavior
     - Parallel loading performance
     - Edge cases (special characters, long IDs, zero counts)
     - Compression and quality preservation
     - High-resolution image handling (4K, 8K)
     - Aspect ratio preservation
   - Updated `FirestoreTests.kt` to initialize Storage emulator and clean up storage files between tests
   - Added storage cleanup utilities for test isolation

   ## Test Plan
   - [x] All ImageRepository integration tests pass (489 lines of test coverage)
   - [x] Storage emulator integration verified
   - [x] Image compression and WebP encoding tested
   - [x] Caching mechanism validated
   - [x] Parallel loading performance tested
   - [x] Edge cases covered (special characters, long IDs, extreme dimensions)
   - [x] High-resolution image handling (up to 12K resolution tests)
   - [x] Aspect ratio preservation verified

   🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: app/src/main/java/com/github/meeplemeet/model/images/ImageRepository.kt [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+import kotlinx.coroutines.withContext
+
+/**
+ * Repository for managing image storage and retrieval. Handles image encoding to WebP format,
+ * caching, and Firebase Storage operations.
+ */
+class ImageRepository {
+  private val storage = FirebaseProvider.storage
+
+  private fun accountPath(id: String) = "accounts/$id"
```
Comment (Atomium90) [inline app/src/main/java/com/github/meeplemeet/model/images/ImageRepository.kt]: For all these paths, even if images are stored in Firebase Storage and collections in Firestore, it could be more intuitive to reuse the collectionName of the referenced objects. That way, it’s immediately clear that the images belong to a given collection.

For example, we could have something like :
`private fun shopPath(id: String) = "$SHOP_COLLECTION_PATH/$id"`

This would make the storage structure easier to understand for someone not familiar with our codebase.

**Review by Atomium90 — COMMENTED** [review]
Overall, the implementation is very good: the generic functions are clear and reusable. The specialized wrappers make usage straightforward and improve readability. Caching images on the local disk is an interesting approach, though it might be worth considering the limits of this strategy if too much data is cached.

That said, I think two aspects could be improved for robustness:
- **Error handling**: failures on disk writes or Firebase operations aren’t caught, which could lead to silent crashes.
- **Use of `Long.MAX_VALUE` in `getBytes`**: this is risky for large files and may cause memory issues. A safer limit or a streaming approach would be preferable.
- A minor point: path naming could be made a bit more intuitive (see the specific comment).

Overall, this is a very good feature addition, backed by a strong test suite.
So I'm leaving this review as a comment rather than an explicit approval/disapproval, since I'd like to discuss those points before deciding if adjustments are necessary.

_Reformulated by Copilot for readabilty._

File: app/src/main/java/com/github/meeplemeet/model/images/ImageRepository.kt (hunk 1 above)
Comment (AntoineBastide47) [reply app/src/main/java/com/github/meeplemeet/model/images/ImageRepository.kt]: Good catch

**Review by Atomium90 — APPROVED** [review]
Thanks for addressing the previous feedback. 

The repository is now more robust against operation failures, and the new path construction feels much more intuitive and maintainable. 

Good work on refining this part. You can merge as soon as all CI checks have passed.

**Review by Atomium90 — CHANGES_REQUESTED** [review]
Overall that's a good addition, the delete functions make the repository more complete and consistent.  

One limitation: for shops and space renters we can only delete all photos at once. It would be useful to also allow deleting a single photo when needed.  

So you can add that and it will be good for me.

**Review by Atomium90 — APPROVED** [review]
Looks good to me. Thanks for that.
